layers/gpuav/spirv/buffer_device_address_pass.cpp

```diff
@@ -169,6 +191,50 @@ bool BufferDeviceAddressPass::Instrument() {
             }
             auto& block_instructions = current_block.instructions_;
 
+            if (module_.settings_.unsafe_mode) {
+                // Pre-Pass optimization where we detect statically all the offsets inside a BDA Struct that are accessed.
+                // From here we can create a range and only do the check once since there is no real way to split a VkBuffer mid
+                // struct.
+                block_struct_range_map_.clear();
+                block_skip_list_.clear();
+                for (auto inst_it = block_instructions.begin(); inst_it != block_instructions.end(); ++inst_it) {
+                    InstructionMeta meta;
+                    if (!RequiresInstrumentation(*function, *(inst_it->get()), meta)) continue;
+
+                    if (!meta.pointer_inst->IsAccessChain()) continue;
+                    // OpAccesschain -> OpLoad/OpBitcast -> OpTypePointer (PSB) -> OpTypeStruct
+                    std::vector<const Instruction*> access_chain_insts;
+
+                    const Instruction* next_inst = meta.pointer_inst;
+                    // First walk back to the outer most access chain
+                    while (next_inst && next_inst->IsAccessChain()) {
+                        access_chain_insts.push_back(next_inst);
+                        const uint32_t access_chain_base_id = next_inst->Operand(0);
+                        next_inst = function->FindInstruction(access_chain_base_id);
+                    }
+                    if (access_chain_insts.empty() || !next_inst) continue;
+
+                    const Type* load_type_pointer = module_.type_manager_.FindTypeById(next_inst->TypeId());
+                    if (load_type_pointer && load_type_pointer->spv_type_ == SpvType::kPointer &&
+                        load_type_pointer->inst_.StorageClass() == spv::StorageClassPhysicalStorageBuffer) {
+                        const Type* struct_type = module_.type_manager_.FindTypeById(load_type_pointer->inst_.Operand(1));
+                        if (struct_type && struct_type->spv_type_ == SpvType::kStruct) {
+                            const uint32_t struct_offset = FindOffsetInStruct(struct_type->Id(), false, access_chain_insts);
+                            if (struct_offset == 0) continue;
+                            uint32_t inst_position = meta.target_instruction->GetPositionIndex();
+                            block_skip_list_.insert(inst_position);
+
+                            Range& range = block_struct_range_map_[struct_type->Id()];
+                            if (struct_offset < range.begin) {
+                                range.begin_instruction = inst_position;
+                                range.begin = struct_offset;
+                            }
+                            range.end = std::max(range.end, struct_offset);
```

Comment:
```suggestion
                            if (struct_offset < range.begin) {
                                range.begin_instruction = inst_position;
                                range.begin = struct_offset;
                            }
                            range.end = std::max(range.end, struct_offset);
```
both `range.begin` and `range.end` can end up equaling `struct_offset`, looks weird? I expected `range.end` to be something like `struct_offset + some_size`

Reply:
its the range of offsets

if you have

```
layout(buffer_reference) buffer BDA {
    uint x[4096];
};

void main() {
    BDA ptr;
    ptr.x[0] = 0
}
```

the Range begin/end would be the same, it works out because we only use the `begin` to really know where the `begin_instruction` is, and we use the `end` to know the actual length/range the check

will add a comment that `begin` here is not actually used other then to find `begin_instruction`